datasets/ru/cbr_banks/crawler.py

```diff
@@ -0,0 +1,189 @@
+            <BicToIntCode xmlns="http://web.cbr.ru/">
+                <BicCode>{bic}</BicCode>
+            </BicToIntCode>
+        </soap:Body>
+    </soap:Envelope>"""
+
+    tree = send_soap_request("BicToIntCode", body)
+
+    # Extract the result
+    result = tree.find(".//ns:BicToIntCodeResult", namespaces=NAMESPACE)
+    if result is not None:
+        return result.text
+    else:
+        print("Result not found in the response")
+        return None
+
+
+def details_by_int_code(internal_code):
+    """Gets detailed credit information for an internal code."""
+    # Create the SOAP request body
+    body = f"""<?xml version="1.0" encoding="utf-8"?>
+    <soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+        <soap:Body>
+            <CreditInfoByIntCodeXML xmlns="http://web.cbr.ru/">
+                <InternalCode>{internal_code}</InternalCode>
+            </CreditInfoByIntCodeXML>
+        </soap:Body>
+    </soap:Envelope>"""
+
+    tree = send_soap_request("CreditInfoByIntCodeXML", body)
+
+    details = {}
+    # Locate the main result element
+    result = tree.find(".//ns:CreditInfoByIntCodeXMLResult", namespaces=NAMESPACE)
+    if result is not None:
+        credit_org_info = result.find("CreditOrgInfo")
+        if credit_org_info is not None:
+            # Extract elements within CO
+            co_data = credit_org_info.find("CO")
+            if co_data is not None:
+                details.update(
+                    {
+                        "reg_number": co_data.findtext("RegNumber"),
+                        "reg_code": co_data.findtext("RegCode"),  # not used
+                        "bic": co_data.findtext("BIC"),
+                        "org_name": co_data.findtext("OrgName"),
+                        "full_name": co_data.findtext("OrgFullName"),
+                        "legal_name": co_data.findtext("csname"),
+                        "en_name": co_data.findtext("encname"),
+                        "phones": co_data.findtext("phones"),
+                        "inc_date": co_data.findtext("DateKGRRegistration"),
+                        "ogrn": co_data.findtext("MainRegNumber"),
+                        "main_reg": co_data.findtext("MainDateReg"),
+                        "address1": co_data.findtext("UstavAdr"),
+                        "address2": co_data.findtext("FactAdr"),
+                        "capital": co_data.findtext("UstMoney"),
+                        "status": co_data.findtext("OrgStatus"),
+                        "ssv_date": co_data.findtext("SSV_Date"),
+                        "lic_withd_num": co_data.findtext("licwithdnum"),
+                        "lic_withd_date": co_data.findtext("licwithddate"),
+                        # "ruleactual": co_data.findtext("ruleactual"),
+                        # "cdmoney": co_data.findtext("cdmoney"),
+                    }
```

Comment:
what do you think of renaming this function to crawl_details, passing the entity in here, and emitting from here rather than `crawl`? Then you don't have to map the xml elements to keys and then again map the keys to props.